File: python/p003.py
# Largest prime factor of 600851475143

n = 600851475143

def isPrime(x):
    for j in range(2, int(x**0.5) + 1):
        if x%j == 0:
            return False
    return True


for i in range(int(n**0.5)+1)[::-1]:
    if n%i == 0:
        if isPrime(i):
            print(i)
            break

File: cpp/p003.cpp
#include <iostream>
#include <cmath>
using namespace std;

int main() {
    long n = 600851475143; // find its largest prime factor
    int dmax = 2;
    for(int d = 2; d < n; d++) {
        if(n%d == 0) {
            dmax = d;
        }
        while(n%d == 0) {
            n /= d; // keep dividing out this factor
        }
    }
    cout << "Largest prime factor of 600851475143: " << n << "\n";
    return 0;
}
